index.js

```diff
@@ -24,20 +24,25 @@ function PolygonLookup( featureCollection ){
 }
 
 /**
- * Find the polygon that a point intersects. Execute a bounding-box search to
+ * Find polygon(s) that a point intersects. Execute a bounding-box search to
  * narrow down the candidate polygons to a small subset, and then perform
  * additional point-in-polygon intersections to resolve any ambiguities.
  *
  * @param {number} x The x-coordinate of the point.
  * @param {number} y The y-coordinate of the point.
- * @return {undefined|object} If one or more bounding box intersections are
- *    found, return the first polygon that intersects (`x`, `y`); otherwise,
- *    `undefined`.
+ * @param {number} [limit] Number of results to return (-1 to return all the results).
+ * @return {undefined|object|array} If one or more bounding box intersections are
+ *    found and limit is undefined, return the first polygon that intersects (`x`, `y`); otherwise,
+ *    `undefined`. If a limit is passed in, return intercecting polygons as an array.
  */
-PolygonLookup.prototype.search = function search( x, y ){
+PolygonLookup.prototype.search = function search( x, y, limit ){
+  if ( limit === 0 ){
+    return [];
+  }
   var bboxes = this.rtree.search( { minX: x, minY: y, maxX: x, maxY: y } );
   var pt = [ x, y ];
-  for( var ind = 0; ind < bboxes.length; ind++ ){
+  var results = [];
+  for( var ind = 0; ind < bboxes.length && ( limit === undefined || limit === -1 || limit > 0 ); ind++ ){
```

Comment:
This is really hard to read and leads to some confusing state later.  Could limit be overridden with `MAX_SAFE_INTEGER` when it's initial value is `-1`?